### ISSUE TITLE
When RAW processing is disabled, already existing JPG versions are also ignored

### ISSUE THREAD

**Issue by iamjackg** [photoview/photoview#677]
**Describe the bug**
When RAW processing is disabled with `PHOTOVIEW_DISABLE_RAW_PROCESSING=1`, Photoview will not scan and display already existing JPG versions of the RAW files.

**To Reproduce**
1. disable RAW processing
2. scan a folder that has RAW+JPEG images

**Expected behavior**
I would expect Photoview to still display photos that already have an existing JPG version, and simply not process additional RAW files _without_ an existing JPG companion. 

**Your environment:**
Feel free to remove and add what you think might be useful.

 - Are you using Docker? **yes**
 - Database **mariadb 10.5**
 - Browser **any**
 - Version `2.3.12 ([0b9d04ef](https://github.com/photoview/photoview/commit/0b9d04eff2b6f896a70e8d081b6706dc4068f732))`

**Comment by kkovaletp** [comment]
@iamjackg, thanks for the report. When you write about the existing JPG version of RAW images, do you refer to the thumbnails and cached images by previous scanner runs, stored in the Photoview cache folder?
If yes, then they are removed because when you disable the RAW processing, the scanner just skips those files as unsupported ones, so later at the DB cleanup stage of the scan job, all not found by the current run files are removed from the DB and cache to free the space and improve DB performance.

Please let me know whether my understanding of your case is correct. Then to implement the improvement, we might need to keep the list of physically existing files in the system (containing all unsupported files as well) to use it as a reference at the cleanup stage and not remove info about previously scanned but not currently supported media.

**Comment by WindLi001** [comment]
@kkovaletp , I found the same problem, and I think I found the reason.

I take photos with camera in RAW+JPEG mode, and then every photo has a RAW version and a JPEG version in the same folder. For example, DSC02392.ARW and DSC02392.JPG. I want the photoview display the JPEG version directly, and ignore the RAW version, so I don't install darktable or Magick, and set the PHOTOVIEW_DISABLE_RAW_PROCESSING=1. And then I find that never the RAW version nor the JPEG version is displayed in photoview.

I had a look at the code, and found this,
api/scanner/media_type/media_type.go:
func (imgType *MediaType) IsSupported() bool {

......

	if executable_worker.Magick.IsInstalled() && imgType.IsRaw() {
		return true
	}

......
}

I think that means the photo with raw version will NOT be displayed if Magick ( in the old version photoview, Darktable) is NOT installed.

I think the code should be changed to 
func (imgType *MediaType) IsSupported() bool {

......

	if imgType.IsRaw() {
		return true
	}

......
}

I will try this in my own environment.

**Comment by kkovaletp** [comment]
@WindLi001 good catch! Thanks for the investigation and analysis.

@googollee, what do you think about this finding? How simple would be the fix to correctly handle all possible cases here?

**Comment by googollee** [comment]
I agree that it's a problem. I'll check it.

**Comment by WindLi001** [comment]
I have tried the fix in my own environment, and I think it works.
Unfortunately, I use a custom branch forked 1 year ago, and I have to sync the commits of main branch in this year before I can merge the fix into main branch. This will take several days.

**Comment by kkovaletp** [comment]
> I have to sync the commits of main branch in this year

Alternatively, you could create a new fork. Just an option)

**Comment by googollee** [comment]
From `EncodeMediaData.EncodeHighRes()` and `ProcessPhotoTask.ProcessMedia()`, it should be possible to convert the side-car JPEG file to a high-resolution caching file. I need to build a environment to check this case.

**Comment by googollee** [comment]
I tried to [CounterpartFilesTask.MediaFound()](https://github.com/photoview/photoview/blob/57d5ac5235fbfe47065af626b43d6a06472a4ed8/api/scanner/scanner_tasks/processing_tasks/counterpart_files_task.go#L22) and don't skip JPEG files if raw process is disabled. With my local test, only returning `false` for skip doesn't work well. It does scan the JPEG file, but it can't stack the raw file with that image. I don't think that's a good solution.

**Comment by googollee** [comment]
Well, it's hard to change than I thought. I hope to say something like:

- If a RAW file with a same-name JPEG file(or any other base types we support) in the same path, we support this file.
- If a RAW file exists alone
  - If `ImageMagick` exists, we support this file.
  - If `ImageMagick` doesn't exist, we don't support this file.

Since checking the same-name file in `counterpart_files_task.go`, far away from `media_type.go`, it's hard to reuse those functions without breaking a hole between packages.

**Comment by googollee** [comment]
Alternative consider:

- We always say we support a RAW file, no matter with/out a JPEG file or with/out ImageMagick. If we can't process that file, we leave a placeholder on the web, maybe with some error messages. 
- Change the ENV to something like `DISABLE_RAW_SUPPORT`, then we always say we don't support a RAW file (but scan the JPEG file).

**Comment by googollee** [comment]
After another thinking, [No RAW in stack](https://github.com/photoview/photoview/issues/677#issuecomment-2388735732) behaves well when disabling RAW processing. As it should not process RAW files, it works with JPEG files directly but not counterpart files.

@kkovaletp Do you agree on this?

**Comment by kkovaletp** [comment]
> * We always say we support a RAW file, no matter with/out a JPEG file or with/out ImageMagick. If we can't process that file, we leave a placeholder on the web, maybe with some error messages

I don't like this option, as it creates a lot of garbage in UI, where end-users will see it. It would make not a good impression

**Comment by kkovaletp** [comment]
> [No RAW in stack](https://github.com/photoview/photoview/issues/677#issuecomment-2388735732) behaves well when disabling RAW processing. As it should not process RAW files, it works with JPEG files directly but not counterpart files.

In general, yes, but could you please describe your solution, adding more details? I feel that I'm missing something)

**Comment by googollee** [comment]
Sure.

- Assuming there is only file: `raw.raf`.

If `PHOTOVIEW_DISABLE_RAW_PROCESSING` is set (`1` or `true`), `photoview` will not process `raw.raf`. No media item in this case.

If `PHOTOVIEW_DISABLE_RAW_PROCESSING` is not set (other values), `photoview` will process `raw.raf`. The original media item is `raw.raf` and stacks with thumbnails generated from `raw.raf`.

- Assuming there are files: `raw.raf` and `raw.jpg`.

If `PHOTOVIEW_DISABLE_RAW_PROCESSING` is set (`1` or `true`), `photoview` will not process `raw.raf` and handle `raw.jpg` as a standalone photo. The original media item is `raw.jpg` and stacks with thumbnails generated from `raw.jpg`.

If `PHOTOVIEW_DISABLE_RAW_PROCESSING` is not set (other values), `photoview` will process `raw.raf` and treat `raw.jpg` as a conter-part file of `raw.raf`. The original media item is `raw.raf` and stacks with `raw.jpg` and thumbnails generated from `raw.jpg`.

**Comment by kkovaletp** [comment]
Great, thanks for the description. So far, I like it.
1 question: what is going to happen if:

> * Assuming there are files: `raw.raf` and `raw.jpg`.
> If `PHOTOVIEW_DISABLE_RAW_PROCESSING` is not set (other values), `photoview` will process `raw.raf` and treat `raw.jpg` as a conter-part file of `raw.raf`. The original media item is `raw.raf` and stacks with `raw.jpg` and thumbnails generated from `raw.jpg`.

But then at some point later an admin sets the `PHOTOVIEW_DISABLE_RAW_PROCESSING` to `1` or` true`, and a new scan job runs

I'd expect the previously scanned RAW files to get updated so that now corresponding counter-part non-RAW files become sources

**Comment by googollee** [comment]
Yeah, that's my question too. From the code, I guess it creates a new item with `raw.raf` and stacks with other jpg files, ignoring the old item. That means the library shows both the old item and the new item. I need to verify this locally.

**Comment by googollee** [comment]
@iamjackg Could you try `ghcr.io/googollee/photoview:process-jpg-677`? It should work well when RAW processing is disabled. Thank you.
